massgov/openmass
# DP-29493: Fix default domain URLs from migration

**Description:**
See ticket


**Jira:** (Skip unless you are MA staff)
DP-29493


**To Test:**
You can see all instances of 'http://default' by taking a mysqldump with `--skip-extended-insert` and parsing with the following, which will return all tables (excluding cache and revision) that contain the default domain, ordered by the number of matches per table.

```
# cat or zcat depending on if the file is gzipped.
zcat recent_prod.sql.gz | grep -v 'INSERT INTO `batch`' | grep -v 'INSERT INTO `cache_entity`' | grep -v 'INSERT INTO `entity_usage`' | grep -v 'INSERT INTO `file_managed`' | grep -v 'INSERT INTO `key_value`' |  grep -v 'INSERT INTO `path_alias`' | grep -v 'INSERT INTO `redirect`' | grep -v 'INSERT INTO `simple_sitemap`' | grep -Ev 'INSERT INTO `.*revision.*`' | grep 'http://default/' | grep -oE '`.*`' | sort | uniq -c | sed 's/`//g' | sort -h -r
```

It may be helpful to take a database export, parse it to see the number of times this domain occurs, run the script with `drush scr modules/custom/mass_migrate/src/fix_default_urls.php`, and take another back up.

You can also check the links on pages. For example, on `/renew-my-license`, the `Check my Application Status` link should reference the entity. You may want to check on the edit page, since GTM is currently rewriting these default URLs.

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)
